debezium-core/src/main/java/io/debezium/processors/reselect/ReselectColumnsPostProcessor.java

```diff
@@ -0,0 +1,240 @@
+        if (requiredColumnSelections.isEmpty()) {
+            LOGGER.debug("No columns require re-selection.");
+            return;
+        }
+
+        final List<String> keyColumns = new ArrayList<>();
+        final List<Object> keyValues = new ArrayList<>();
+        for (org.apache.kafka.connect.data.Field field : key.schema().fields()) {
+            keyColumns.add(field.name());
+            keyValues.add(key.get(field));
+        }
+
+        Map<String, Object> selections = null;
+        try {
+            final String reselectQuery = jdbcConnection.buildReselectColumnQuery(tableId, requiredColumnSelections, keyColumns, source);
+            selections = jdbcConnection.reselectColumns(reselectQuery, tableId, requiredColumnSelections, keyValues);
+            if (selections.isEmpty()) {
+                LOGGER.warn("Failed to find row in table {} with key {}.", tableId, key);
+                return;
+            }
+        }
+        catch (SQLException e) {
+            LOGGER.warn("Failed to re-select row for table {} and key {}", tableId, key, e);
+            return;
+        }
+
+        // Iterate re-selection columns and override old values
+        for (Map.Entry<String, Object> selection : selections.entrySet()) {
+            final String columnName = selection.getKey();
+            final Column column = table.columnWithName(columnName);
+            final org.apache.kafka.connect.data.Field field = after.schema().field(columnName);
+
+            final Object convertedValue = getConvertedValue(column, field, selection.getValue());
+            if (LOGGER.isTraceEnabled()) {
+                LOGGER.trace("Replaced field {} value {} with {}", field.name(), value.get(field), convertedValue);
+            }
+            after.put(field.name(), convertedValue);
+        }
+    }
+
+    @Override
+    public void setConnectorConfig(CommonConnectorConfig connectorConfig) {
+        if (connectorConfig instanceof RelationalDatabaseConnectorConfig) {
+            final RelationalDatabaseConnectorConfig config = (RelationalDatabaseConnectorConfig) connectorConfig;
+            this.unavailableValuePlaceholder = new String(config.getUnavailableValuePlaceholder());
+            this.unavailableValuePlaceholderBytes = config.getUnavailableValuePlaceholder();
+        }
+    }
+
+    @Override
+    public void setValueConverter(ValueConverterProvider valueConverter) {
+        this.valueConverterProvider = valueConverter;
+    }
+
+    @Override
+    public void setDatabaseConnection(JdbcConnection connection) {
+        this.jdbcConnection = connection;
+    }
+
+    @Override
+    public void setDatabaseSchema(RelationalDatabaseSchema schema) {
+        this.schema = schema;
+    }
+
+    private List<String> getRequiredColumnSelections(TableId tableId, Struct after) {
+        final List<String> columnSelections = new ArrayList<>();
+        for (org.apache.kafka.connect.data.Field field : after.schema().fields()) {
```

Comment:
I think here the logic needs more thinking. I'd expect user to filter in/out columns depending whether they are interested in re-selection. SO maybe we need addtional config options and rename `columns.list` option?

Reply:
In that case, would something as simple as `column.include.list` and `column.exclude.list` suffice?

Reply:
Yes, and the `column.list` should be renamed to something like `reselct.null.columns` or similar to be more descriptive.